scripts/main/view.js

```diff
@@ -25,35 +32,43 @@ view.albums = {
 	},
 
 	content: {
+		/** @returns {void} */
 		init: function () {
 			let smartData = "";
 			let albumsData = "";
 			let sharedData = "";
 
 			// Smart Albums
-			if (albums.json.smartalbums != null) {
+			// TODO: My IDE complains that `smart_albums` is always non-null.
+			// If albums has not yet been loaded, then the *parent* object
+			// `albums.json` is null, but if `albums.json` exists, then
+			// `albums.json.smart_albums` exists too.
+			// However, `albums.json.smart_albums` might be empty if the
+			// use is un-authenticated and no public smart album exists.
+			// But being empty is something different than being `null`.
```

Comment:
Well, there's one case I can think of: search. It results in a bastardized `albums.json` that has only `albums`. I hate that code and always wanted to rip it out, but it's there.

Reply:
Good point. I will ensure to put your answer into a comment when I proceed with adopting the front-end to the new JSON API. This answer is the kind of information which is really helpful when someone tries to understand the code.